backend/infrahub/proposed_change/tasks.py

```diff
@@ -644,3 +648,144 @@ def _should_render_artifact(artifact_id: str | None, managed_branch: bool, impac
+
+    file_info = extract_repo_file_information(
+        full_filename=commit_worktree.directory / generator_definition.file_path,
+        repo_directory=repository.directory_root,
+        worktree_directory=commit_worktree.directory,
+    )
+    generator_instance = await _define_instance(model=model, service=service)
+
+    check_message = "Instance successfully generated"
+    try:
+        log.debug(f"repo information {file_info}")
+        log.debug(f"Root directory : {repository.directory_root}")
+        generator_class = generator_definition.load_class(
+            import_root=repository.directory_root, relative_path=file_info.relative_repo_path_dir
+        )
+
+        generator = generator_class(
+            query=generator_definition.query,
+            client=service.client,
+            branch=model.branch_name,
+            params=model.variables,
+            generator_instance=generator_instance.id,
+            convert_query_response=generator_definition.convert_query_response,
+            infrahub_node=InfrahubNode,
+        )
+        generator._init_client.request_context = model.context.to_request_context()
+        await generator.run(identifier=generator_definition.name)
+        generator_instance.status.value = GeneratorInstanceStatus.READY.value
+    except ModuleImportError as exc:
+        conclusion = ValidatorConclusion.FAILURE
+        generator_instance.status.value = GeneratorInstanceStatus.ERROR.value
+        check_message = f"Failed to import generator: {exc.message}"
+        log.exception(check_message, exc_info=exc)
+    except Exception as exc:
+        conclusion = ValidatorConclusion.FAILURE
+        generator_instance.status.value = GeneratorInstanceStatus.ERROR.value
+        check_message = f"Failed to execute generator: {str(exc)}"
+        log.exception(check_message, exc_info=exc)
+
+    log.info("Generator run completed, starting update")
+    await generator_instance.update(do_full_update=True)
+
+    check = None
+    existing_check = await service.client.filters(
+        kind=InfrahubKind.GENERATORCHECK, validator__ids=model.validator_id, instance__value=generator_instance.id
+    )
+    if existing_check:
+        check = existing_check[0]
+
+    if check:
+        check.created_at.value = Timestamp().to_string()
+        check.conclusion.value = conclusion.value
+        await check.save()
+    else:
+        check = await service.client.create(
+            kind=InfrahubKind.GENERATORCHECK,
+            data={
+                "name": model.target_name,
+                "origin": model.repository_id,
+                "kind": "GeneratorDefinition",
+                "validator": model.validator_id,
+                "created_at": Timestamp().to_string(),
+                "message": check_message,
+                "conclusion": conclusion.value,
+                "instance": generator_instance.id,
+            },
+        )
+        await check.save()
```

Comment:
Not related to this PR but this pattern is repeated in a number of places, i.e

1. Search of a check
2. Update if it exists
3. Create a new check if it doesn't

We have something similar in start_validator() we should probably clean this up and remove some of the boilerplate.